### ISSUE TITLE
Block update recursion causes a crash when xdebug is enabled

### ISSUE THREAD

**Issue by thebigsmileXD** [pmmp/PocketMine-MP#251]
### Issue description

Death loops when breaking huge cacti

### Steps to reproduce the issue

1. Build a high cactus
2. Break the lowest block

### OS and versions

* PocketMine-MP: https://github.com/pmmp/PocketMine-MP/commit/67940ad6ab4b9632823a749a9e7fa973d4500620
* PHP:
* Server OS:
* Game version: PE/Win10 (delete as appropriate)

### Crashdump, backtrace or other files

Gc too many cycles

**Comment by dktapps** [comment]
Cannot reproduce, but related issues:
- Immense lag
- Far too many dropped cacti
- Drops freeze in midair

**Comment by thebigsmileXD** [comment]
Should i provide a video as proof?

**Comment by SOF3** [comment]
We don't need **proof**. It is to everyone's least concern whether the bug exists or not if it cannot be properly identified and fixed.

**Comment by thebigsmileXD** [comment]
Nice, so because you can't find the bug you say it doesn't exist?
![image](https://cloud.githubusercontent.com/assets/8733998/21892781/be70173e-d8d8-11e6-812c-2ae336ad76e0.png)

**Comment by dktapps** [comment]
Do read his comment again. We don't need proof the bug exists; we need a way we can reproduce the bug. As I have already stated, I am unable to reproduce this bug. Until it can be reproduced, it cannot be fixed unless there is an obvious cause, in which case reproducing is still preferable anyway, to ensure that any changes actually _fix_ the issue.

**Comment by thebigsmileXD** [comment]
By the way, same happens if you have a giant row of wall signs and break one.

**Comment by SOF3** [comment]
* [Cactus.php:76](https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/block/Cactus.php#L76) Cactus::onUpdate()
* [Level.php:1594](https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/level/Level.php#L1594) Level::useBreakOn()
* [Block.php:359](https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/block/Block.php#L359) Block::onBreak()
* [Level.php:1445](https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/level/Level.php#L1445) Level::setBlock()
* [Level.php:1028](https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/level/Level.php#L1028) Level::updateAround()

Recursion to initial step.

Theoretically reproduced issue.
However, whether deep stack recursion should be prevented remains an issue.

**Comment by dktapps** [comment]
Vertically it isn't really an issue due to the build height limit. If this could happen horizontally however, that would be a whole different ball-game.

**Comment by dktapps** [comment]
This is only reproducible **with xdebug** with a nesting limit set. The following 5 calls are recursed over for every block in the stack.
```
  850.4678   38003688 107. pocketmine\block\Cactus->onUpdate() C:\Users\Admin\Documents\GitHub\pocketmine-mp\src\pocketmine\level\Level.php:1001
  850.4678   38003880 108. pocketmine\level\Level->useBreakOn() C:\Users\Admin\Documents\GitHub\pocketmine-mp\src\pocketmine\block\Cactus.php:76
  850.4679   38005400 109. pocketmine\block\Block->onBreak() C:\Users\Admin\Documents\GitHub\pocketmine-mp\src\pocketmine\level\Level.php:1569
  850.4679   38005592 110. pocketmine\level\Level->setBlock() C:\Users\Admin\Documents\GitHub\pocketmine-mp\src\pocketmine\block\Block.php:361
  850.4683   38005688 111. pocketmine\level\Level->updateAround() C:\Users\Admin\Documents\GitHub\pocketmine-mp\src\pocketmine\level\Level.php:1420
```

**Comment by SOF3** [comment]
Note that PHP normally doesn't impose limit on function nesting level, until it goes to something around 65536 levels (as I have tested with PocketMine PHP Linux binaries from get.pmmp.io), or around 322536 levels (as I have tested with PocketMine PHP Windows binaries from bintray). xdebug by default imposes a restriction of 256 nesting levels. Previous PHP binaries distributions from PocketMine (probably many years ago) might have also imposed a restriction of 100 nesting levels.
